provisioning-telephony/nodejs/index.js

```diff
@@ -0,0 +1,331 @@
+            console.error(err);
+        });
+}
+
+// Create location function 
+function createLocation() {
+    const body = {
+        name: inputTemplate.location.name, // ex: My Location Name
+        emergencyNumber: {
+            number: inputTemplate.location.number, // ex: +639XXXXXXXXX
+            type: 'default',
+        },
+        address: {
+            street1: inputTemplate.location.street1,
+            city: inputTemplate.location.city,
+            state: inputTemplate.location.state,
+            zipcode: inputTemplate.location.zipcode,
+            country: inputTemplate.location.country,
+            countryFullName: inputTemplate.location.countryFullName,
+        },
+    };
+    locationsApi.postLocations(body)
+        .then((data) => {
+            console.log(`postLocations success! data: ${JSON.stringify(data, null, 2)}`);
+            console.log('Location successfully created!');
+            locationInfo = data;
+            getEdgeSite();
+        })
+        .catch((err) => {
+            console.log('There was a failure calling postLocations');
+            console.log(err);
+        });
+}
+
+// This function will get the default Edge Group - PureCloud Voice - AWS
+function getEdgeSite() {
+    const opts = {
+        pageSize: 25,
+        pageNumber: 1,
+        sortBy: 'name',
+        sortOrder: 'ASC',
+    };
+    // Get list of virtual edges
+    telephonyProvidersEdgeApi.getTelephonyProvidersEdgesSites(opts)
+        .then((data) => {
+            console.log(`getTelephonyProvidersEdgesSites success! data: ${JSON.stringify(data, null, 2)}`);
+            // Find default PureCloud Voice - AWS provider edge on the list
+            const awsItem = data.entities.find((entitiesItem) => entitiesItem.name === 'PureCloud Voice - AWS');
+            createSite(awsItem);
+        })
+        .catch((err) => {
+            console.log('There was a failure calling getTelephonyProvidersEdgesSites');
+            console.error(err);
+        });
+}
+
+// This function will create a site using the location value and edge group value saved to a variable earlier
+function createSite(awsItem) {
+    // Format current date to make default edgeAutoUpdateConfig's start date and end date
+    const today = new Date();
+    const dateConfig = `${today.getFullYear()}-${today.getMonth() + 1}-${today.getDate()}`;
+    const body = {
+        name: inputTemplate.site.name, // Ex: My Site Name
+        primarySites: [{
+            id: awsItem.id,
+            name: awsItem.name,
+            selfUri: awsItem.selfUri,
+        }],
+        secondarySites: [{
+            id: awsItem.id,
+            name: awsItem.name,
+            selfUri: awsItem.selfUri,
+        }],
+        edgeAutoUpdateConfig: {
+            timeZone: inputTemplate.site.timeZone, // Ex: America/New_York
+            rrule: 'FREQ=DAILY',
+            start: `${dateConfig}T02:00:00.000`,
```

Comment:
Why is this hardcoded to these specific times? Should briefly mention why or what to set this to in the create site step.

Reply:
2AM-5AM is the current default value. I will mention that in the yaml content. i also modified the code to:

    let startTime = today.toISOString(today.setHours(10,0,0,0)).substring(0,23); // default value is 2AM
    let endTime =  today.toISOString(today.setHours(13,0,0,0)).substring(0,23); // default value is 5AM